src/AspNetCoreModuleV2/AspNetCore/src/proxymodule.cxx

```diff
@@ -83,96 +84,104 @@ ASPNET_CORE_PROXY_MODULE::OnExecuteRequestHandler(
     REQUEST_NOTIFICATION_STATUS retVal = RQ_NOTIFICATION_CONTINUE;
     IAPPLICATION* pApplication = NULL;
     STACK_STRU(struFileName, 256);
-
-    if (g_fInShutdown)
+    try
```

Comment:
Why not wrap the entire function in try catch?

Reply:
returns REQUEST_NOTIFICATION_STATUS and has to set 500 response code.